src/api.js

```diff
@@ -4,14 +4,22 @@ const apiClient = axios.create({
   timeout: 10000,
 });
 
-export function getArticles(currPage) {
-  return apiClient.get(`/articles?limit=10&p=${currPage}`).then(({ data }) => {
+export function getArticles(currPage, topicName) {
+  let requestUrl = `articles?p=${currPage}`;
+  if (topicName) {
+    requestUrl += `&topic=${topicName}`;
+  }
```

Comment:
This can be handled with the axios second argument. See the NC notes about this, but if you pass `{params:{topicName:"myfavetopic"}}` the axios takes care of checking if there is topic name and passing it to the request.